services/app-api/src/emailer/emailer.ts

```diff
@@ -46,31 +46,37 @@ type Emailer = {
     sendCMSNewPackage: (
         submission: LockedHealthPlanFormDataType,
         submissionName: string,
```

Comment:
Do you think it would work to just stop passing `submissionName` and `rateName` both to each emailer function and instead just pass the `programs`. One parameter instead of 2? I can't tell why we should keep passing in strings instead of just generating the strings within the emailer functions closer to where they are used.

Reply:
Yea totally doable. I think the reason why I didn't go this way is it seems like repeat code in each emailer function, like how it used to with generating rate name in the emailer functions. Also now that I changed `store.findPrograms()` to find combined `programIDs` and `rateProgramIDs` we should be able to pass that to the emailers. This was the reason for the bug you mentioned below.

Reply:
@macrael Can you weigh in here? I know you had thoughts about what the parameters and bounds of the emailer service should be

Reply:
I noticed a few things here

1. rateName is being added as a new param to all the email functions, even though some of those functions wrap up the submissionName inside of an options dictionary and some don’t. Ideally we’d be more consistent. Some of the functions take the whole submission and some take an object like “UpdatedEmailData” with just the info we need for this particular email. Until we sort all of that out though, let’s add new options the way the function currently accepts them. 
2. Thinking a little ahead, we are soon going to have packages that have multiple rates associated with them. A single “rateName” will not cut it. I don’t know if an array of rate names will work, but my hunch is we’ll want to be displaying info about each rate individually and just indexing into an array of names feels a bit haphazard. 
3. Setting the rateName to undefined for ContractOnly submissions definitely looks a bit off to me, I don’t love it when a funciton can be called with arguments that don’t actually make sense. Like if you passed a rate name in even though the submission is ContractOnly, what would happen?

This seems like it might be worth a quick discussion to talk about the pro/cons of how we’re passing info into the emailer.

Reply:
> Thinking a little ahead, we are soon going to have packages that have multiple rates associated with them A single “rateName” will not cut it.

This is a very good point I overlooked.

> Setting the rateName to undefined for ContractOnly submissions definitely looks a bit off to me, I don’t love it when a funciton can be called with arguments that don’t actually make sense.

Also a good point

> stop passing submissionName and rateName both to each emailer function and instead just pass the programs.

@haworku suggestion seems like the better way to go? We can generate names as needed in the emailers, especially when we're dealing with multiple rates per package.

> This seems like it might be worth a quick discussion to talk about the pro/cons of how we’re passing info into the emailer.

This would be good.